### PR TITLE
Fix & enable 'ColorKeyword' and 'ColorVariable' scss-lint cops

### PR DESCRIPTION
**Pull request by joshlory**
I replaced SCSS color literals with existing variables where possible.  For colors that need to be specific values I created new variables.

### CODE REVIEW

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 1]
```diff
@@ -156,7 +156,7 @@ img.video_thumbnail {
   float: left;
   clear: left;
   width: 23px;
-  border: 1px solid lightgray;
+  border: 1px solid $light_gray;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:160]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 2]
```diff
@@ -1842,7 +1842,7 @@ div.gallery_activity {
   img {
     width: $img_size;
     height: $img_size;
-    border: $img_border_size groove lightgray;
+    border: $img_border_size groove $light_gray;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:1790]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 3]
```diff
@@ -1866,13 +1866,13 @@ div.mini_gallery_activity {
...
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 4]
```diff
@@ -2121,7 +2121,7 @@ $modal-no-icon-gap: 22px;
     height: auto;
     margin: 0 15px 0 0;
     float: left;
-    border: lightgrey;
+    border: $light_gray;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:2069]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 5]
```diff
@@ -1918,7 +1918,7 @@ $information-desktop: "only screen and (min-width: 1024px)";
   .answerbutton {
     &.lock-answers {
       pointer-events: none;
-      color: grey;
+      color: $light_gray;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:1866]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss [hunk 6]
```diff
@@ -1,6 +1,8 @@
+@import "color";
+
 body {
-  background-color: #fff;
-  color: #333;
+  background-color: $white;
+  color: $dark_charcoal;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss [hunk 7]
```diff
@@ -11,7 +13,7 @@ p, ol, ul, td {
 }
 
 pre {
-  background-color: #eee;
+  background-color: $lightest_gray;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss:16]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss [hunk 8]
```diff
@@ -23,38 +25,38 @@ div {
...
 
 #error_explanation {
   width: 450px;
-  border: 2px solid #c00;
+  border: 2px solid $red;
   padding: 7px;
   padding-bottom: 0;
   margin-bottom: 20px;
-  background-color: #f0f0f0;
+  background-color: $white;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss:50]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss [hunk 9]
```diff
@@ -23,38 +25,38 @@ div {
 }
 
 #notice {
-  color: green;
+  color: $realgreen;
 }
 
 .field_with_errors label {
-  color: red;
+  color: $red;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss:32]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 10]
```diff
@@ -1469,7 +1469,7 @@ a.download-video {
     vertical-align: top;
   }
   .error_in_field {
-    color: red;
+    color: $red;
```
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:1417]: Snapped to a different (but similar) color in the palette.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss (hunk 6 above)
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss]: @tanyaparker is this the correct color for default text?

Comment (tanyaparker) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss]: Yup!

File: shared/css/color.scss [hunk 11]
```diff
@@ -54,12 +54,17 @@ $inset_color: $lighter_gray;
 $dark_color:  $purple;
 $hdr_color:   $purple;
 
-// some extra colors
+// Some extra colors.
 $red: #c00;
-$realyellow: yellow;
+$realgreen: #008000;
```
Comment (tanyaparker) [inline shared/css/color.scss:59]: Should we be consistent about using 6 or 3 digit shortcuts?

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 12]
```diff
@@ -2501,10 +2501,10 @@ $width: 400px;
...
   top: 0;
   right: 0;
-  color: white;
+  color: $white;
   background-color: $light_purple;
   padding: 3px;
   z-index: 100;
   cursor: pointer;
-  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.75);
+  box-shadow: 0 0 7px 2px $shadow;
```
Comment (ashercodeorg) [inline dashboard/app/assets/stylesheets/application.css.scss:2454]: The definition of `$shadow` below is `rgba(0, 0, 0, 0.3);`. Is this change intentional?

Also, is the change from `0 0 10px 0` to `0 0 7px 2px` intentional?

**Comment by ashercodeorg** [comment]
LGTM.

**Comment by breville** [comment]
no additional comments beyond those already here.  lgtm.  thanks for doing this.

File: dashboard/app/assets/stylesheets/application.css.scss (hunk 12 above)
Comment (joshlory) [inline dashboard/app/assets/stylesheets/application.css.scss:2454]: Yup!  This is just the environment tag:

screen shot 2016-08-09 at 10 13 49 am
I tweaked the metrics a little bit in order to re-use the existing shadow color.

Comment (ashercodeorg) [inline dashboard/app/assets/stylesheets/application.css.scss:2454]: Got it. Should the next DOTD expect an eyes test change from this?

File: shared/css/color.scss (hunk 11 above)
Comment (joshlory) [inline shared/css/color.scss:59]: We currently follow ['HexLength' style: short](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#hexlength) but we could switch to always 6 for consistency.  Note that `#008000` can't be shortened.

Do you prefer 6 hex digits for all colors?

Comment (tanyaparker) [inline shared/css/color.scss:59]: My mistake, I misread. I don't feel strongly either way, keeping short style works with me.

File: dashboard/app/assets/stylesheets/scaffolds.css.scss (hunk 6 above)
Comment (joshlory) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss]: I think `$dark_charcoal` might be too light, and sacrifice readability on bad monitors:

Before:

screen shot 2016-08-09 at 3 56 21 pm

After:

screen shot 2016-08-09 at 3 56 28 pm

Comment (tanyaparker) [inline dashboard/app/assets/stylesheets/scaffolds.css.scss]: Gotcha. Agree!
